src/containers/Review/assignment/ReviewLevel.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect } from 'react'
+import { Dropdown, Label } from 'semantic-ui-react'
+import { useLanguageProvider } from '../../../contexts/Localisation'
+import { AssignmentDetails, LevelDetails, Filters, FullStructure } from '../../../utils/types'
+
+export const ALL_LEVELS = 0
+
+type ReviewLevelProps = {
+  filters: Filters | null
+  setFilters: (filters: Filters) => void
+  structure: FullStructure
+  assignments: AssignmentDetails[]
+}
+
+const ReviewLevel: React.FC<ReviewLevelProps> = ({
+  assignments,
+  structure,
+  filters,
+  setFilters,
+}) => {
+  const { strings } = useLanguageProvider()
+  const { getLevelsOptions } = useHelpers()
+
+  useEffect(() => {
+    setFilters({
+      selectedLevel: ALL_LEVELS,
+      selectedStage: structure.info.current.stage.number, // Fixed with current for application
+    })
+  }, [])
+
+  const changeFilters =
+    (filterType: keyof Filters) =>
+    (_: any, { value }: any) => {
+      if (filters) setFilters({ ...filters, [filterType]: value })
+    }
+
+  if (!filters) return null
+
+  const stage = structure.stages.find(({ stage: { number } }) => number === filters.selectedStage)
+
+  if (!stage) return null
+
+  return (
+    <div className="section-single-row-box-container" id="review-filters-container">
+      <div className="centered-flex-box-row">
+        <Label className="simple-label" content={strings.REVIEW_FILTER_SHOW_TASKS_FOR} />
+        <Dropdown
+          className="reviewer-dropdown"
+          options={getLevelsOptions(stage.levels)}
+          value={filters?.selectedLevel}
+          onChange={changeFilters('selectedLevel')}
+        />
+      </div>
+    </div>
+  )
+}
+
+const useHelpers = () => {
```

Comment:
Does this need to be a hook rather than just a static function? Is it just because you have to use "strings" in there, in which case it has to be a hook to call another hook?

I've come across this issue too quite a bit -- I've just been calling the "useLanguageProvider" at the top level and passing in the strings object to static functions as a parameter. I guess either way is fine, don't suppose you have any idea about the performance differences?

Reply:
Not sure... I only kept it there since it was already being used. But it could be a function inside the component I guess...

Reply:
All good, it's fine. Just something to keep in mind for future. For some reason I'm reluctant to create new hooks unless they're really necessary -- but I don't really have any justification for that 😆